### PR TITLE
Add metadata for October 2021 patches

### PR DESCRIPTION
**Pull request by mfielding**
Adding patch metadata for Oracle's October 2021 PSUs, including 12.1,
12.2, and 19c.  11.2.0.4 and 18c no longer receive patchset updates.

The patch directory layout for GI/RDBMS combo patches (under
rdbms_patches) has changed, with an additional layer of nesting.  So
patch_siubdir now contains two directory levels.  To identify the
correct path, loko at bundle.xml and identify the singleton patch that
has all four components.  In case of multiple matches, one subdirectory
should be significantly larger; take that one.

### CODE REVIEW

File: roles/common/defaults/main.yml [hunk 1]
```diff
@@ -296,6 +298,7 @@ gi_patches:
 # Warning: GIRU 32126842 for 19.10.0.0.210119 has critical regressions and password protected. Development recommends 19.11.0.0.210420 Patch 32545008 instead of the JAN 2021 quarter RU
 #  - { category: "RU", base: "19.3.0.0.0", release: "19.10.0.0.210119", patchnum: "32126842", patchfile: "p32126842_190000_Linux-x86-64.zip", patch_subdir: "/32226239", prereq_check: FALSE, method: "opatchauto apply", ocm: FALSE, upgrade: FALSE, md5sum: "opnaS/8dS13yGypZCCN8WA==" }
 #  - { category: "RU", base: "19.3.0.0.0", release: "19.11.0.0.210420", patchnum: "32578973", patchfile: "p32578973_190000_Linux-x86-64.zip", patch_subdir: "/32545008", prereq_check: FALSE, method: "opatchauto apply", ocm: FALSE, upgrade: FALSE, md5sum: "YLEOruyjCOdDvUOMBUazNQ==" }
+  - { category: "RU", base: "19.3.0.0.0", release: "19.9.0.0.211019", patchnum: "33248471", patchfile: "p33248471_190000_Linux-x86-64.zip", patch_subdir: "/33192694", prereq_check: FALSE, method: "opatchauto apply", ocm: FALSE, upgrade: FALSE, md5sum: "3LgQa6fZ/UHs3XZ8GvtwrQ==" }
```
Comment (jcnars) [inline roles/common/defaults/main.yml:301]: Verification of patch directory shows: 

```bash
.
├── 33248471
│   ├── 33182768   <== this is the GI RU, some molecules go into only GI_HOME, some into only RDBMS_HOME & some into both
│   │   ├── 32585572  <== goes into only GI_HOME
│   │   ├── 33192793  <== goes into only DB_HOME for non-RAC, both homes for RAC
│   │   ├── 33208107  <== goes only into GI_HOME
│   │   ├── 33208123  <== goes into both homes
│   │   ├── 33239955  <== goes only into GI_HOME
│   │   ├── automation
│   │   ├── bundle.xml
│   │   ├── README.html
│   │   └── README.txt
│   ├── 33192694   <============ this is the OJVM RU, that goes only into the RDBMS home
│   │   ├── etc
│   │   ├── files
│   │   ├── README.html
│   │   └── README.txt
│   ├── PatchSearch.xml
│   └── README.html
└── PatchSearch.xml
```

So, the `patch_subdir: "/33192694"` under `gi_patches:` should actually be `patch_subdir: "/33182768"`, like so:
```
gi_patches:
...
  - { category: "RU", base: "19.3.0.0.0", release: "19.13.0.0.211019", patchnum: "33248471", patchfile: "p33248471_190000_Linux-x86-64.zip", patch_subdir: "/33182768", prereq_check: FALSE, method: "opatchauto apply", ocm: FALSE, upgrade: FALSE, md5sum: "3LgQa6fZ/UHs3XZ8GvtwrQ==" }
```

And for `rdbms_patches:`, it should not be: `patch_subdir: "/33182768/33192793"`, instead, it should read: `patch_subdir: "/33192694"`, like so:
```
  - { category: "RU_Combo", base: "19.3.0.0.0", release: "19.13.0.0.211019", patchnum: "33248471", patchfile: "p33248471_190000_Linux-x86-64.zip", patch_subdir: "/33192694", prereq_check: TRUE, method: "opatch apply", ocm: FALSE, upgrade: TRUE, md5sum: "3LgQa6fZ/UHs3XZ8GvtwrQ==" }

```

And also, a typo:
Change `release: "19.9.0.0.211019"` to `release: "19.13.0.0.211019"` under both `rdbms_patches:` and `gi_patches:` data structures.

Comment (jcnars) [reply roles/common/defaults/main.yml:301]: I have submitted the corrected version for 19c metadata in [pull 104](https://github.com/google/bms-toolkit/pull/104).

You may discard the 19c changes patch metadata here and proceed to correct the 12c / 18c patch metadata in this PR.

Following may be an useful reference:
https://github.com/google/bms-toolkit/blob/master/docs/patch-metadata.md

**Comment by jcnars** [comment]
Triggering the prowjob here before merging [PR 106.](https://github.com/google/bms-toolkit/pull/106)
/test bms-toolkit-install

We expect this:
* to be queued (another prowjob running now, triggered from: https://github.com/google/bms-toolkit/pull/106#issuecomment-1146732460) 
* and then when this prowjob's turn comes (triggered with `/test bms-toolkit-install` above),  the prowjob should fail with same error as @mfielding received above.

Once we hit the expected job failure here, I will merge [PR 106.](https://github.com/google/bms-toolkit/pull/106) into master branch and trigger the same job here, at which point we expect the job to run and actually test the commits proposed in this PR 102.

**Comment by jcnars** [comment]
It is queued: https://screenshot.googleplex.com/67MzKL3GQ6ecsiM

**Comment by jcnars** [comment]
Good, it failed: https://oss.gprow.dev/view/gs/bmaas-testing-prow-bucket/pr-logs/pull/google_bms-toolkit/102/bms-toolkit-install/1533302493205762048 , as expected.

Now merging the PR 106 into mainline.

**Comment by jcnars** [comment]
Triggering again, after merging PR 106: 

/test bms-toolkit-install

This should run the tests with the changes in this PR 102.

**Comment by jcnars** [comment]
Yup, it went past the previous point of error...from the logs at: https://oss.gprow.dev/view/gs/bmaas-testing-prow-bucket/pr-logs/pull/google_bms-toolkit/102/bms-toolkit-install/1533312184329703424
```
/home/prow/go/src/github.com/google/bms-toolkit
Command used:
./cleanup-oracle.sh --ora-version 19 --inventory-file /etc/files_needed_for_tk/nonrac-inv --yes-i-am-sure --ora-disk-mgmt udev --ora-swlib-path /u01/oracle_install --ora-asm-disks /etc/files_needed_for_tk/nonrac-asm.json --ora-data-mounts /etc/files_needed_for_tk/nonrac-datamounts.json
Running with parameters from command line or environment variables:
INVENTORY_FILE=/etc/files_needed_for_tk/nonrac-inv
ORA_ASM_DISKS=/etc/files_needed_for_tk/nonrac-asm.json
ORA_DATA_MOUNTS=/etc/files_needed_for_tk/nonrac-datamounts.json
ORA_DISK_MGMT=udev
ORA_ROLE_SEPARATION=TRUE
ORA_STAGING=/u01/oracle_install
ORA_SWLIB_PATH=/u01/oracle_install
ORA_VERSION=19.3.0.0.0
Ansible params: 
Found Ansible: ansible-playbook is /usr/local/bin/ansible-playbook
Running Ansible playbook: ansible-playbook -i /etc/files_needed_for_tk/nonrac-inv  brute-cleanup.yml
PLAY [dbasm] *******************************************************************
TASK [Gathering Facts] *********************************************************
ok: [linuxserver44.orcl]
TASK [Verify that Ansible on control node meets the version requirements] ******
ok: [linuxserver44.orcl] => {
    "changed": false,
    "msg": "Ansible version is 2.9.27.post0, continuing"
}
TASK [brute-ora-cleanup : Shut down RAC clusterware] ***************************
skipping: [linuxserver44.orcl]
TASK [brute-ora-cleanup : Deconfigure RAC CRS] *********************************
skipping: [linuxserver44.orcl]
TASK [brute-ora-cleanup : Stop services with srvctl] ***************************
```

**Comment by mfielding** [comment]
@jcnars PTAL;  tests are succeeding.  the extra commits here are from a rebase to current master;  I can create a brand-new PR if that makes it easier to review.

**Comment by mfielding** [comment]
/retest

Got an error on /u01 being in use when creating the filesystem.  But when I re-run mkfs manually, it succeeds.

**Comment by mfielding** [comment]
Will close this in favor of PR #113

**Pull request closed without merging**